### DISCUSSION TITLE
Is there a place where you can put executables for rish?

### DISCUSSION THREAD

**Discussion by Quackdoc** [RikkaApps/Shizuku#446] [Q&A]
I have some apps I want to try and run with rish on A10, however due to android's permission system I can't launch them from just anywhere. Is there a location we can place executables so we can launch them with rish? or are we limited to what the phone ships with?

EDIT: I am aware you can copy apps to `/data/local/tmp` and execute them from there temporarily but I'm looking for something more permanent

**Comment by Knud3** [comment]
You can use `/data/data/com.termux/files/usr/bin` (or `../usr/bin` from home) with Termux.

**Reply by Quackdoc** [reply]
Due to permissions system (and just a lack of PATH handling) Im not sure how viable that is. How would I execute them?

**Reply by Knud3** [reply]
Just `chmod +x /data/data/com.termux/files/usr/bin/foo` and execute with `foo`.

**Reply by Quackdoc** [reply]
with rish? that will give permission denied, at least on A10

**Reply by Knud3** [reply]
What are you trying to do? What app are you trying to run?

**Reply by Quackdoc** [reply]
In this case an `top` alternative called bottom. cross compiled, everything works fine, Trying to run it with rish need to find a place I can out the binary and run it from

**Reply by Knud3** [reply]
How about semi-permanent solution with `rish -c 'cp /storage/emulated/0/foo/bar /data/local/tmp'` to `/data/data/com.termux/files/usr/etc/bash.bashrc`

**Reply by Knud3** [reply]
You might be able to also use it directly with `rish -c 'sh /storage/emulated/0/foo/bar'`
